### PR TITLE
Add property State in additional to ProvisioningState

### PR DESCRIPTION
**Pull request by shihying**
Add property State in additional to ProvisioningState
We introduced State as suggested by ARM-Architect to provide more custom state outside of provisioning.

Currently value set for State would be same as ProvisioiningState until service side make further change.

## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx).

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md).**
- [x] **If changes were made to any cmdlet, the XML help was regenerated using the [platyPSHelp module](https://github.com/Azure/azure-powershell/blob/dev/documentation/platyPSHelp-documentation.md).**
- [x] **If any large changes are made to a service, they are reflected in the respective [change log](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#updating-the-change-log).**

### [General Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] The pull request does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/dev/documentation/breaking-changes.md) (unless a major version change occurs in the assembly and module).

### [Testing Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.
- [x] PowerShell scripts used in tests should do any necessary setup as part of the test or suite setup, and should not use hard-coded values for locations or existing resources.

### [Cmdlet Signature Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#cmdlet-signature-guidelines)
- [x] New cmdlets that make changes or have side effects should implement `ShouldProcess` and have `SupportShouldProcess=true` specified in the cmdlet attribute. You can find more information on `ShouldProcess` [here](https://gist.github.com/markcowl/338e16fe5c8bbf195aff9f8af0db585d#what-is-the-change).
- [x] Cmdlet specifies `OutputType` attribute if any output is produced - if the cmdlet produces no output, it should implement a `PassThru` parameter.

### [Cmdlet Parameter Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#cmdlet-parameter-guidelines)
- [x] Parameter types should not expose types from the management library - complex parameter types should be defined in the module.
- [x] Complex parameter types are discouraged - a parameter type should be simple types as often as possible. If complex types are used, they should be shallow and easily creatable from a constructor or another cmdlet.
- [x] Cmdlet parameter sets should be mutually exclusive - each parameter set must have at least one mandatory parameter not in other parameter sets.

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@shihying__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (shic). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: src/ResourceManager/AnalysisServices/Commands.AnalysisServices.Test/ScenarioTests/AsTests.ps1 [hunk 1]
```diff
@@ -95,13 +98,15 @@ function Test-AnalysisServicesServer
 		Suspend-AzureRmAnalysisServicesServer -ResourceGroupName $resourceGroupName -Name $serverName
 		[array]$serverGet = Get-AzureRmAnalysisServicesServer -ResourceGroupName $resourceGroupName -Name $serverName
 		$serverGetItem = $serverGet[0]
+		Assert-True {$serverGetItem.State -like "Paused"}
```
Comment (athipp) [inline src/ResourceManager/AnalysisServices/Commands.AnalysisServices.Test/ScenarioTests/AsTests.ps1]: Add a comment saying that this is to ensure backward compatibility. The breaking server changes will be done after this is released.

Comment (shihying) [inline src/ResourceManager/AnalysisServices/Commands.AnalysisServices.Test/ScenarioTests/AsTests.ps1]: done.

Comment (shihying) [inline src/ResourceManager/AnalysisServices/Commands.AnalysisServices.Test/ScenarioTests/AsTests.ps1]: done

**Comment by shihying** [comment]
@athipp 
AS module version bumped 
Change log updated

**Comment by shihying** [comment]
power shell admin : please do not merge until SDK repo merge PR for this chagne

**Comment by shihying** [comment]
+@markcowl
+@shahabhijeet

**Comment by markcowl** [comment]
@shihying you must publish your new library and update the nuget package reference for your cmdlets

**Comment by shihying** [comment]
@markcowl 
Library 1.0.2-preview is published, package reference is updated already., thanks

File: src/ResourceManager/AnalysisServices/ChangeLog.md [hunk 2]
```diff
@@ -18,6 +18,8 @@
         - Additional information about change #1
 -->
 ## Current Release
+* Added State property in additional to ProvisioningState
```
Comment (markcowl) [inline src/ResourceManager/AnalysisServices/ChangeLog.md]: You need to be more specific about this change. Which cmdlets will be affected, how would the user use the new State property.  How is it different than Provinsioning State?

Comment (shihying) [inline src/ResourceManager/AnalysisServices/ChangeLog.md]: @markcowl , I have added more information about this change.
Currently ASPAAS service still return same for both State and ProvisioningState, we could not tell the user the actual different example until service side make change in future.

I already provided the difference in intention

**Comment by shihying** [comment]
@markcowl @shahabhijeet
Can you take a look on my updated ChangeLog.md? thanks

**Comment by shihying** [comment]
@markcowl @shahabhijeet
Can you take a look on my updated ChangeLog.md? thanks

**Comment by shahabhijeet** [comment]
@azuresdkci add to whitelist

**Comment by shihying** [comment]
@shahabhijeet  @markcowl  ping....

**Comment by cormacpayne** [comment]
@markcowl @shahabhijeet ping

**Comment by shahabhijeet** [comment]
@shihying If the service is not ready with providing more info for the State property, is there a point in adding this property before the service enables it.
Also what is the time frame for the service team to enable this property from server side?
is it happening in 1 month... 6 months?

**Comment by shihying** [comment]
@shahabhijeet 
The service side already provide this property in the response but the value would be same as ProvisioningState - I think we want to let the powershell to be compatible with latest service response :)

There is no exact plan yet - but I think it must happen in 1month..6months..thanks

**Comment by markcowl** [comment]
on demand run: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/1395/

LGTM once the build passes

**Pull request merged**